src/torrent.cpp

```diff
@@ -3317,11 +3314,11 @@ namespace libtorrent
 				continue;
 
 #if TORRENT_USE_I2P
-			if (r.i2pconn && boost::algorithm::ends_with(i->hostname, ".i2p"))
+			if (r.i2pconn && string_ends_with(i->hostname, ".i2p"))
 			{
-				// this is an i2p name, we need to use the sam connection
+				// this is an i2p name, we need to use the same connection
```

Comment:
this should actually be sam, or maybe better "SAM" (to make it more obvious that it's an acronym)